### PR TITLE
Add input validation to role name field in console role create wizard

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

https://github.com/wso2/identity-apps/assets/41533942/dc410404-9933-422c-8907-6d3c0dc6e7aa

### Related Issues
- https://github.com/wso2/product-is/issues/19051

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx [hunk 1]
```diff
@@ -63,11 +53,13 @@ const CreateConsoleRoleWizardBasicInfoForm: FunctionComponent<CreateConsoleRoleW
             helperText={ (
                 <Hint>
                     Provide a distinctive and meaningful display name for the role. This name should clearly represent
-                    the purpose or responsibilities associated with this role within the Console application. E.g.{ " " }
-                    <Code>application:read</Code>
+                    the purpose or responsibilities associated with this role within the Console application.
+                    E.g.{ " " }<Code>application:read</Code>
```
Comment (Achintha444) [inline apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx:57]: Ideally this should have localization.

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx [hunk 2]
```diff
@@ -63,11 +53,13 @@ const CreateConsoleRoleWizardBasicInfoForm: FunctionComponent<CreateConsoleRoleW
...
                 </Hint>
             ) }
             placeholder="Enter Role name"
+            maxLength={ 10 }
```
Comment (Achintha444) [inline apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx:61]: Add this to a constant file.

**Review by Achintha444 — DISMISSED** [review]

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx (hunk 1 above)
Comment (pavinduLakshan) [reply apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx:57]: Yes, but this was not added from this PR. This should be fixed in a separate effort [1].

[1] https://github.com/wso2/identity-apps/pull/4747#issuecomment-1820144558

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx]: These validation props are not required here. Therefore removed with 2f40fd5

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx:57]: Ack

File: apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/console-settings/components/console-roles/create-console-role-wizard/create-console-role-wizard-basic-info-form.tsx]: Ack

**Review by Achintha444 — APPROVED** [review]

**Comment by pavinduLakshan** [comment]
This improvement will be added with the changes done for https://github.com/wso2/product-is/issues/19052, hence closing the PR.

cc: @JeethJJ

**Pull request closed without merging**